gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/inmemory/PolicyValidationDomainServiceInMemory.java

```diff
@@ -15,12 +15,30 @@
  */
 package inmemory;
 
+import io.gravitee.apim.core.plugin.model.PolicyPlugin;
 import io.gravitee.apim.core.policy.domain_service.PolicyValidationDomainService;
+import io.gravitee.apim.core.policy.exception.UnexpectedPolicyException;
+import io.gravitee.definition.model.v4.ApiType;
+import io.gravitee.rest.api.service.exceptions.InvalidDataException;
+import java.util.List;
 
 public class PolicyValidationDomainServiceInMemory implements PolicyValidationDomainService {
 
     @Override
     public String validateAndSanitizeConfiguration(String policyName, String configuration) {
+        if (policyName.contains("throw_invalid_data_exception")) {
```

Comment:
With the InMemory classes, we have the idea of a classical database, but in memory, despite in this class you have a specific case: a new business rule which is implemented here. That will be confusing for other tests using this class in the future.

Maybe you can consider creating a fake instead, as we did here: https://github.com/gravitee-io/gravitee-api-management/blob/fa91c9231d4b9e8b1741a8d4d41e7d55f7580338/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/fakes/FakeAnalyticsQueryService.java

it has the advantage of letting the tester set fields of the fake directly and just expect the behavior he wants for his test case.

Reply:
Hmmm well when it's a domain service I don't think of a repository. juste class in memory. (I saw it for another domain service so I didn't ask myself any questions)
but I'm really fine calling it Fake. like `FakePolicyValidationDomainService`